packages/composite-checkout/src/components/checkout.js

```diff
@@ -195,6 +195,7 @@ PaymentMethodsStep.propTypes = {
 function BillingDetailsStep( { isActive, isComplete, setStepNumber, onChangeBillingContact } ) {
 	const localize = useLocalize();
 	const [ paymentData, setPaymentData ] = usePaymentMethodData();
+	const [ isDomainContactVisible, setIsDomainContactVisible ] = useState( false );
```

Comment:
🤔 This is the common parent of the two `BillingContactComponent` components... but by putting `isDomainContactVisible` here we're bringing the concept of "domain contact information" into the package. `BillingFields ` is in the package too of course, so it's not really a problem but it makes me wonder: what if we later add another boolean we need to track? Do we keep adding them here? That's not ideal.

I had some ideas to keep the state either inside `BillingFields` or to put it in `paymentData` which I've just added as another comment. Let me know what you think.

Reply:
yeah, I had this thought in the back of my mind when I added it here. We're probably better off dealing with it locally inside the `BillingFields` component.